python/minetester/minetest_env.py

```diff
@@ -0,0 +1,505 @@
+        self.context = zmq.Context()
+        self.socket = self.context.socket(zmq.REQ)
+        self.socket.connect(f"tcp://localhost:{self.env_port}")
+
+    def _reset_minetest(self):
+        # Determine log paths
+        reset_timestamp = datetime.datetime.now().strftime("%m-%d-%Y,%H:%M:%S")
+        log_path = os.path.join(
+            self.log_dir,
+            f"{{}}_{reset_timestamp}_{self.unique_env_id}.log",
+        )
+
+        # Close Mintest processes
+        if self.client_process:
+            self.client_process.kill()
+
+        # (Re)start Minetest client
+        self.client_process = self._start_minetest_client(
+            log_path,
+        )
+
+    def _perform_client_handshake(self):
+        # handshake is an empty action
+        pb_action = remoteclient_capnp.Action.new_message()
+        self.socket.send(pb_action.to_bytes())
+
+    def _check_world_dir(self):
+        if self.world_dir is None:
+            raise RuntimeError(
+                "World directory was not set. Please, provide a world directory "
+                "in the constructor or seed the environment!",
+            )
+
+    def _delete_world(self):
+        if self.world_dir and os.path.exists(self.world_dir):
+            shutil.rmtree(self.world_dir, ignore_errors=True)
+
+    def _check_config_path(self):
+        if self.config_path is None:
+            raise RuntimeError(
+                "Minetest config path was not set. Please, provide a config path "
+                "in the constructor or seed the environment!",
+            )
+
+    def _delete_config(self):
+        if os.path.exists(self.config_path):
+            os.remove(self.config_path)
+
+    def _write_config(self):
+        config = dict(
+            # Base config
+            mute_sound=True,
+            show_debug=False,
+            enable_client_modding=True,
+            csm_restriction_flags=0,
+            enable_mod_channels=True,
+            screen_w=self.display_size[0],
+            screen_h=self.display_size[1],
+            fov=self.fov_y,
+            # Adapt HUD size to display size, based on (1024, 600) default
+            hud_scaling=self.display_size[0] / 1024,
+            # Experimental settings to improve performance
+            server_map_save_interval=1000000,
+            profiler_print_interval=0,
+            active_block_range=2,
+            abm_time_budget=0.01,
+            abm_interval=0.1,
+            active_block_mgmt_interval=4.0,
+            server_unload_unused_data_timeout=1000000,
+            client_unload_unused_data_timeout=1000000,
+            full_block_send_enable_min_time_from_building=0.0,
+            max_block_send_distance=100,
+            max_block_generate_distance=100,
+            num_emerge_threads=0,
+            emergequeue_limit_total=1000000,
+            emergequeue_limit_diskonly=1000000,
+            emergequeue_limit_generate=1000000,
+            game_dir="games/minetest_game",
```

Comment:
separate from performance related settings with a comment line for clarity

Reply:
This shouldn't have been hardcoded. I've changed it now to either provide game_dir or world_dir as the argument, and pass the game_dir via config file instead of specifying --gameid on the CLI